types/react-navigation/index.d.ts

```diff
@@ -476,7 +476,7 @@ export interface LayoutEvent {
 }
 
 interface NavigationContainerProps {
-  navigation: NavigationProp<any, NavigationAction>
+  navigation?: NavigationProp<any, NavigationAction>
```

Comment:
As i mentioned above, the navigation prop does not need to be provided every time a StackNavigator is instantiated. Perhaps it would be better suited to have a default value, instead of optional.